apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -802,39 +783,13 @@ func (api *ProvisionerAPI) SetInstanceInfo(ctx context.Context, args params.Inst
 	}
 	setInstanceInfo := func(arg params.InstanceInfo) error {
 		tag, err := names.ParseMachineTag(arg.Tag)
-		if err != nil {
+		if err != nil || !canAccess(tag) {
 			return apiservererrors.ErrPerm
 		}
-		machine, err := api.getMachine(canAccess, tag)
-		if err != nil {
-			return err
-		}
-		volumes, err := storagecommon.VolumesToState(arg.Volumes)
-		if err != nil {
-			return err
-		}
-		volumeAttachments, err := storagecommon.VolumeAttachmentInfosToState(arg.VolumeAttachments)
-		if err != nil {
-			return err
-		}
```

Comment:
This should have had a `TODO(storage): ...` comment.

Reply:
Indeed, sorry I missed this one.